include/SampleDecoder.h

```diff
@@ -53,6 +53,8 @@ class SampleDecoder
 	static auto decode(const QString& audioFile) -> std::optional<Result>;
 	static auto supportedAudioTypes() -> std::vector<AudioType>;
 
+	inline static auto s_supportedAudioTypes = supportedAudioTypes();
```

Comment:
I think I'd prefer doing it the way I suggested with the local static variable in `supportedAudioTypes()`. It's simple and keeps stuff out of the header.

If you cache it using a data member, you shouldn't use "s_" for something you're using as a public constant. If you want to expose the constant for external use, it should be: 
```c++
static inline const auto SupportedAudioTypes = supportedAudioTypes();
```
And the supportedAudioTypes() method should be private.

Reply:
> I think I'd prefer doing it the way I suggested with the local static variable in supportedAudioTypes(). It's simple and keeps stuff out of the header.

I prefer doing it your way as well. I initially was hesitant as I thought it was working on the same vector multiple times for each call to that function, but I see that you adding an if-guard against that. I will change it to what you did soon as it's a lot simpler and keeps the header file clean, just working on the translation issue right now.